packages/select/src/components/base-select/Component.tsx

```diff
@@ -279,6 +280,9 @@ export const BaseSelect = forwardRef(
             );
         }, [multiple, selectedItems, disabled, name, handleNativeSelectChange, options, menuProps]);
 
+        const needRenderOptionsList =
+            flatOptions.length > 0 || Boolean(optionsListEmptyPlaceholder);
```

Comment:
Не совсем понял логику, тут случайно не так должно быть?
```suggestion
            flatOptions.length > 0 || !optionsListEmptyPlaceholder;
```